src/inputManager.cpp

```diff
@@ -0,0 +1,187 @@
+#include "inputManager.hpp"
+
+#include "logger.hpp"
+
+InputManager* InputManager::sInstance = nullptr;
+
+void InputManager::initialize() {
+    sInstance = new InputManager();
+}
+
+void InputManager::quit() {
+    delete sInstance;
+}
+
+void InputManager::reset() {
+    sInstance->internalReset();
+}
+
+void InputManager::registerObject(GameObject* obj, InputType inputType) {
+    sInstance->internalRegisterObject(obj, inputType);
+}
+
+void InputManager::updateInput() {
+    sInstance->internalUpdateInput();
+}
+
+InputState InputManager::getInputState(SDL_Keycode key) {
+    return sInstance->internalGetInputState(key);
+}
+
+InputState InputManager::getInputState(GameObject* obj, SDL_Keycode key) {
+    return sInstance->internalGetInputState(obj, key);
+}
+
+InputState InputManager::getInputState(InputKey key) {
+    return sInstance->internalGetInputState(key);
+}
+
+InputState InputManager::getInputState(GameObject* obj, InputKey key) {
+    return sInstance->internalGetInputState(obj, key);
+}
+
+Vec2 InputManager::getMouseLocation() {
+    return sInstance->internalGetMouseLocation();
+}
+
+bool InputManager::ctrlDown() {
+    return sInstance->internalCtrlDown();
+}
+
+bool InputManager::altDown() {
+    return sInstance->internalAltDown();
+}
+
+bool InputManager::shiftDown() {
+    return sInstance->internalShiftDown();
+}
+
+InputManager::InputManager() {
+    for (int i = 0; i < SDL_NumJoysticks(); ++i) {
+        if (SDL_IsGameController(i)) {
```

Comment:
Can do an early return for readability:
`if (!SDL_IsGameController(i)) continue;`

Reply:
Done